### PR TITLE
[IPHLPAPI] Fix getNumRoutes() return value, on Mib error

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

This code was added as is in r8194.

## Proposed changes

- Obviously not `status`. Not current `numRoutes` either, I assume.

### CODE REVIEW

**Comment by SergeGautherie** [comment]
Trivial commit comment update.

**Comment by learn-more** [comment]
So to fix a return value, you add 2 goto statements?

**Comment by SergeGautherie** [comment]
> you add 2 goto statements

Yes, in hope "to prevent issues like CORE-14353 and this one".
If you don't want that part, simply/explicitly say so. I am suggesting only.

File: dll/win32/iphlpapi/ipstats_reactos.c [hunk 1]
```diff
@@ -387,17 +386,18 @@ DWORD getNumRoutes(void)
             memset( &isnmp, 0, sizeof( isnmp ) );
             status = tdiGetMibForIpEntity( tcpFile, &entitySet[i], &isnmp );
             if( !NT_SUCCESS(status) ) {
-                tdiFreeThingSet( entitySet );
-                closeTcpFile( tcpFile );
-                return status;
+                numRoutes = 0;
+                goto NoMib;
```
Comment (gedmurphy) [inline dll/win32/iphlpapi/ipstats_reactos.c]: Put tdiGetEntityIDSet back as it was, set numRoutes = 0 and just use a break statement to exit the loop.

File: dll/win32/iphlpapi/ipstats_reactos.c [hunk 2]
```diff
@@ -387,17 +386,18 @@ DWORD getNumRoutes(void)
...
             }
             numRoutes += isnmp.ipsi_numroutes;
         }
     }
 
     TRACE("numRoutes: %d\n", (int)numRoutes);
 
+NoMib:
```
Comment (gedmurphy) [inline dll/win32/iphlpapi/ipstats_reactos.c]: There's no need for any labels

File: dll/win32/iphlpapi/ipstats_reactos.c [hunk 3]
```diff
@@ -377,8 +377,7 @@ DWORD getNumRoutes(void)
 
     if( !NT_SUCCESS(status) ) {
         TRACE("failure: %08x\n", (int)status );
-        closeTcpFile( tcpFile );
-        return 0;
+        goto NoEntitySet;
```
[the remark this answers is not in the document]
Comment (SergeGautherie) [reply dll/win32/iphlpapi/ipstats_reactos.c]: Reverted.

File: dll/win32/iphlpapi/ipstats_reactos.c (hunk 1 above)
Comment (SergeGautherie) [reply dll/win32/iphlpapi/ipstats_reactos.c]: Agreed. Done.

File: dll/win32/iphlpapi/ipstats_reactos.c (hunk 2 above)
Comment (SergeGautherie) [reply dll/win32/iphlpapi/ipstats_reactos.c]: Reverted.

File: dll/win32/iphlpapi/ipstats_reactos.c [hunk 4]
```diff
@@ -387,17 +386,18 @@ DWORD getNumRoutes(void)
             memset( &isnmp, 0, sizeof( isnmp ) );
             status = tdiGetMibForIpEntity( tcpFile, &entitySet[i], &isnmp );
             if( !NT_SUCCESS(status) ) {
```
Comment (SergeGautherie) [inline dll/win32/iphlpapi/ipstats_reactos.c]: Added a `WARN`.

**Comment by gedmurphy** [comment]
Code looks good.
Is there a Jira case to back this up, and does it fix any tests?

**Comment by SergeGautherie** [comment]
> Is there a Jira case to back this up, and does it fix any tests?

None that I am aware of: done by code review only.
Trying it on testbots could apply.

**Pull request merged**